osbrain/agent.py

```diff
@@ -88,20 +87,16 @@ def __init__(self, name=None, host=None):
         self.context = zmq.Context()
         self.poller = zmq.Poller()
 
-        # This in-process socket could, eventually, handle safe access to
-        # memory from other threads (i.e. when using Pyro proxies).
-        socket = self.context.socket(zmq.REP)
-        address = 'inproc://loopback'
-        socket.bind(address)
-        self.register(socket, address, 'loopback', self._handle_loopback)
+        # A loopback socket where, for example, timers are processed
+        self.bind('REP', alias='loopback', addr='loopback',
+                  handler=self._handle_loopback, transport='inproc',
+                  serializer='pickle')
```

Comment:
Not really. It is an internal socket (no need to use `raw`) and it passes complex objects/types (such as those passed when using timers) which would be hard to serialize otherwise.